packages/core/src/components/ImageOverlay.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { useContext, useState } from 'react'
+import { ExamComponentProps } from '../createRenderChildNodes'
+import { getNumericAttribute, mapChildElements } from '../dom-utils'
+import { faEye } from '@fortawesome/free-solid-svg-icons/faEye'
+import { faEyeSlash } from '@fortawesome/free-solid-svg-icons/faEyeSlash'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import ResponsiveMediaContainer from './ResponsiveMediaContainer'
+import { CommonExamContext } from './CommonExamContext'
+import classNames from 'classnames'
+
+function ImageOverlay(props: ExamComponentProps) {
+  const { element } = props
+  const [sliderValues, setSliderValue] = useState<number[]>(
+    Array.from({ length: element.children.length }, (_, i) => (i === 0 ? 1 : 0))
+  )
+
+  const imageElements = mapChildElements(element, (childElement) => childElement)
+
+  const onSetSliderValue = (event: React.ChangeEvent<HTMLInputElement>, index: number) => {
+    const updatedSliders = [...sliderValues]
+    updatedSliders[index] = Number(event.target.value)
+    setSliderValue(updatedSliders)
+  }
+
+  return (
+    <div className="e-image-overlay e-mrg-b-1">
+      <div className="e-columns e-columns--center-h">
+        <div>
+          {sliderValues.map((sliderValue, index) => (
```

Comment:
IMO nää kaks mappia olis vielä ehkä vähän selkeämpiä, jos mäpättäis elementin childNodeja eikä opacityjä, koska XML:n rakenne on alkuperäinen "totuuden" lähde ja opacityt on johdettu siitä. Ei myöskään tarvitsisi tuota imageElementsiä ollenkaan.

Myöskin `{...props}`-spreadin sijasta `renderChildElementsin` vois antaa Sliderille eksplisiittisesti - nyt lukijan pitää tajuta että propsit passataan vaan `renderChildElementsin` takia ja että propseissa jo oleva `element` tulee myös muistaa ylikirjoittaa!

Reply:
Eksplisiittisyydessä edustatte nyt Eeron kanssa eri koulukuntaa! :) Mutta mennään sun ehdotuksen mukaan, näin se oli myös alunperin.